### DISCUSSION TITLE
HEIC photos rotated when they shouldn't

### DISCUSSION THREAD

**Discussion by joachimtingvold** [photoprism/photoprism#2717] [Q&A] [answered]
Hi,

I recently imported a bunch of HEIC photos taken on an iPhone 13 Pro, where many of them are rotated when they shouldn't. The converted HEIC-image (.jpg) and the thumb is rotated 180 degrees. The Live Photo video is not rotated (and is correct orientation).

Looking at the EXIF data, I can see the following;

```
Orientation                     : Rotate 180
Rotation                        : 180
```

As per the EXIF data, the rotation done by photoprism is technically "correct", but when I'm opening the original HEIC image in multiple different image viewers (Photoshop, GIMP, etc), the image is correctly orientated. Looking at the picture on Apple systems (iPhone, iPad, apple.com, Photos.app, Preview.app), the image is also correctly orientated.

How come these other apps/systems show the picture correctly orientated, despite the "rotate 180" EXIF data? Also, why is only converted HEIC + thumb affected in photoprism? (and not Live Photo/video).

These rotations happen on 220902-62aba3200, so it does not seem to be related to #2214 .

Yet another scenario where #464 would be really useful.

**Comment by joachimtingvold** [comment, marked as answer]
Samples sent to `samples@photoprism.app` with subject `Photo sample for discussion #2717`.

**Reply by lastzero** [reply]
It would be great if you could help us test "heif-convert"! We suspect that it partly depends on the underlying Linux distribution. More information and a test guide can be found here:

- https://github.com/photoprism/photoprism/issues/1064#issuecomment-1250149075
- https://docs.photoprism.app/developer-guide/media/heif/#testing-conversion-and-orientation

**Comment by joachimtingvold** [comment]
The `irot` box also seems to have the "rotate 180 degrees" info;

```
| | | Box: irot -----
| | | size: 9   (header size: 8)
| | | rotation: 180 degrees (CCW)
```

**Comment by joachimtingvold** [comment]
I think I have found the culprit; HEIC-conversion rotates the image (according to EXIF/irot attributes), but it also copies the EXIF-data, causing browsers et al to double-rotate (since it tries to rotate an already rotated image).

I'm not sure what the proper solution here would be. Either HEIC conversion should not rotate (and let the browser/client do the rotation), or if it does rotate, it should change EXIF rotation attributes accordingly.

If I remove the EXIF data from the photoprism-converted .jpg-image via `exiftool -all= <filename>.jpg`, the image has correct orientation.

```
jocke@keklolwtf: ~/Downloads $ exiftool 20220918_115636_58145C13.heic  | grep -i rot
Orientation                     : Rotate 180
Rotation                        : 180
jocke@keklolwtf: ~/Downloads $ heif-info 20220918_115636_58145C13.heic -d | grep irot -A2
| | | Box: irot -----
| | | size: 9   (header size: 8)
| | | rotation: 180 degrees (CCW)
jocke@keklolwtf: ~/Downloads $ exiftool 20220918_115636_58145C13.heic.jpg  | grep -i rot
Orientation                     : Rotate 180
```

**Comment by adutta98** [comment]
I still have this behavior on a docker based implementation of Photoprism on Unraid. Live video is rotated 180 degrees from iPhone 13 pro max. Just wondering why is not tagged as a bug for resolution?

**Reply by lastzero** [reply]
Does it work on our demo? What version do you use? Did you try our preview build yet? Any other helpful information eg custom config options?

**Reply by barry-luijten** [reply]
After installing 231011, I also noticed the wrongly rotated live videos for about 50-75% of my live photos (iPhone 12 Pro), while the photo itself is showing correctly oriented. 
It seems that photos in portrait-orientation **always** show the video 90 degrees rotated, while photos in landscape-orientation show the video 180 degrees rotated (upsidedown) only when the photo is detected/marked as being 180 degrees rotated (in files tab of the editor).

**Reply by lastzero** [reply]
@barry-luijten How did you install 231011? Do you use our official Docker image?

**Reply by barry-luijten** [reply]
@lastzero I'm using the official docker images on Docker Swarm. I updated the existing stack using Portainer.
I must add that I'm not using HEIC files, but jpg+mov files. I hope I'm not trespassing this discussion, but the stated problem sounded exactly like mine 😉

**Reply by lastzero** [reply]
Orientation of mov files is indeed something completely different then converting HEIC to JPEG. If those movs are transcoded, then the question is why ffmpeg doesn't keep the orientation. If they are played directly, then the question is why browsers don't show them as specified. For both, I would be thankful if you could do some research!

**Reply by barry-luijten** [reply]
@lastzero I'm sorry. it seems to be a problem related to my Edge browser (version 118). Chrome (also version 118) on Windows and Safari on iOS show the videos correctly oriented. Also when I download the mov and play it with Windows Movies app, it shows correctly. So this probably not a Photoprism issue. Thanks for you reply though; I never suspected this could be a browser problem! :)
